docs/concepts/activation.md

```diff
@@ -0,0 +1,321 @@
+which is to put you into a subshell.
+When you activate this way your existing shell is paused and you're put into a
+new one configured by Flox.
+Once this shell exits (via `exit` or `Ctrl-D`, for example), your original shell
+is resumed and your are put back into it.
+
+```d2 scale="1.0"
+shape: sequence_diagram
+user_shell: User shell
+subshell: Subshell
+
+user_shell -> subshell: flox activate
+subshell -> subshell: do work in the subshell
+subshell -> user_shell: exit
+```
+
+### In-place
+
+This method is similar to the first in that you're still in an interactive
+shell,
+but in this case it's the original shell.
+
+To make this happen you could do one of these options in Bash:
+```bash
+# Option 1
+source <(flox activate)
+
+# Option 2
+eval "$(flox activate)"
+```
+
+In both cases Flox emits a script that configures the shell,
+and the shell executes that code to configure itself.
+
+```d2 scale="1.0"
+shape: sequence_diagram
+user_shell: User shell
+user_shell -> user_shell: source <(flox activate)
+user_shell."Back to you"
+```
+
+In order to configure a `default` environment that's activated for every new
+shell,
+you would put a line like this in your `.bashrc`, `.zshrc`, `.tcshrc`, or
+`config.fish`.
+You could do this manually, but Flox will also prompt you to do it for you
+the first time you attempt to install a package in a directory without an
+environment and with no environments currently active.
+
+### Command
+
+Sometimes you just want to run a command in the context of your environment,
+maybe because you have some tools available in your environment that aren't
+available outside the environment.
+
+You could do this in a subshell:
+- Enter the subshell via `flox activate`
+- Run the command
+- Exit the subshell via `exit`
+
+You could do a similar thing with an in-place activation:
+- Configure your existing shell with `source <(flox activate)`.
+- Run the command
+- Your shell is still configured by Flox when you're done
+
+That's a lot of ceremony to run that one command though,
+and the in-place activation leaves the environment still activated in your
+shell,
+which you may not want.
+
+The easy way to do this is:
+```bash
+$ flox activate -- <your command>
+```
+
+This starts a Flox-configured subshell, runs your command,
+and immediately exits to put you back into your shell.
+
+```d2 scale="1.0" pad="1"
+shape: sequence_diagram
+user_shell: User shell
+subshell: Subshell
+user_shell -> subshell: "flox activate -- cmd"
+subshell -> subshell: run "cmd"
+subshell -> user_shell: automatic exit
+user_shell."Back to you"
+```
+
+## Activation flow
+
+In order to understand where `hook` and `profile` fit into the picture,
+we need to explore the timeline of what an activation looks like.
+Much of this is dictated by what gets inherited when you create subshells.
+Don't feel like you need to understand this entirely in order to use Flox,
+it's just here to help you if you want a deeper understanding.
```

Comment:
I like this. I think this page gives people the information they need to really understand the details of what's going on if they want, but it offers that information in a way that doesn't get in the way of people who just need a surface level understanding